coordinator/history/history.go

```diff
@@ -0,0 +1,228 @@
+		return nil, errors.New("mismatch between hashSize and hash function size")
+	}
+	return h, nil
+}
+
+// ConfigureSigningKey sets the signing key for validation and signing of the protected history parts.
+func (h *History) ConfigureSigningKey(signingKey *ecdsa.PrivateKey) {
+	h.signingKey = signingKey
+}
+
+// GetManifest returns the manifest for the given hash.
+func (h *History) GetManifest(hash [hashSize]byte) ([]byte, error) {
+	return h.getContentaddressed("manifests/%s", hash)
+}
+
+// SetManifest sets the manifest and returns its hash.
+func (h *History) SetManifest(manifest []byte) ([hashSize]byte, error) {
+	return h.setContentaddressed("manifests/%s", manifest)
+}
+
+// GetPolicy returns the policy for the given hash.
+func (h *History) GetPolicy(hash [hashSize]byte) ([]byte, error) {
+	return h.getContentaddressed("policies/%s", hash)
+}
+
+// SetPolicy sets the policy and returns its hash.
+func (h *History) SetPolicy(policy []byte) ([hashSize]byte, error) {
+	return h.setContentaddressed("policies/%s", policy)
+}
+
+// GetTransition returns the transition for the given hash.
+func (h *History) GetTransition(hash [hashSize]byte) (Transition, error) {
+	transitionBytes, err := h.getContentaddressed("transitions/%s", hash)
+	if err != nil {
+		return Transition{}, err
+	}
+	var transition Transition
+	if err := transition.unmarshalBinary(transitionBytes); err != nil {
+		return Transition{}, fmt.Errorf("unmarshaling transition: %w", err)
+	}
+	return transition, nil
+}
+
+// SetTransition sets the transition and returns its hash.
+func (h *History) SetTransition(transition Transition) ([hashSize]byte, error) {
+	return h.setContentaddressed("transitions/%s", transition.marshalBinary())
+}
+
+// GetLatest returns the verified transition for the given hash.
+func (h *History) GetLatest() (LatestTransition, error) {
+	if h.signingKey == nil {
+		return LatestTransition{}, errors.New("signing key not configured")
+	}
+	transitionBytes, err := h.store.Get("transitions/latest")
+	if err != nil {
+		return LatestTransition{}, fmt.Errorf("getting latest transition: %w", err)
+	}
+	var latestTransition LatestTransition
+	if err := latestTransition.unmarshalBinary(transitionBytes); err != nil {
+		return LatestTransition{}, fmt.Errorf("unmarshaling latest transition: %w", err)
+	}
+	if err := latestTransition.verify(&h.signingKey.PublicKey); err != nil {
+		return LatestTransition{}, fmt.Errorf("verifying latest transition: %w", err)
+	}
+	return latestTransition, nil
+}
+
+// SetLatest signs and sets the latest transition if the current latest is equal to oldT.
+func (h *History) SetLatest(oldT, newT LatestTransition) error {
+	if h.signingKey == nil {
+		return errors.New("signing key not configured")
+	}
+	if err := newT.sign(h.signingKey); err != nil {
+		return fmt.Errorf("signing latest transition: %w", err)
+	}
+	if err := h.store.CompareAndSwap("transitions/latest", oldT.marshalBinary(), newT.marshalBinary()); err != nil {
```

Comment:
How do we set the first `latest`? If I pass an empty `oldT`, it marshals to 64 zero bytes, but the `CompareAndSwap` compares it with an empty slice.

Reply:
Any ideas?

Reply:
Accept `oldT` as pointer and treat `nil` as "must be empty"?